### PR TITLE
Store placemark URLs

### PR DESCRIPTION
**Pull request by ronplaymeta**
the way this PR works is similar to PR 2 description of this https://github.com/bldrs-ai/Share/issues/599
and additionally, user can drop place mark in scene by shift + mouse left button clicking, not only mouse left button double-clicking.

[test url](https://deploy-preview-657--bldrs-share.netlify.app/share/v/gh/ronplaymeta/bldrs-ai-Share/main/public/index.ifc#f:placemark)

### CODE REVIEW

**Comment by ronplaymeta** [comment]
I discussed with @pablo-mayrgundter, and made a new branch from [PR 672.](https://github.com/bldrs-ai/Share/pull/672)

**Comment by pablo-mayrgundter** [comment]
Hi Ron, good to hear.  Could you update the description?  If it's the same as the PR#2 step exactly, just refer to that, or note any differences.

Please include a description of how to demo the feature, and please use a different example repo than the main page one.

**Comment by ronplaymeta** [comment]
thank you for the detailed feedback on PR 657.
currently, the app can only use one git repo and now it's yours.
if we change git repo, the contents of notes will be changed to changed repo, shall i proceed then?
and would you please merge PR 672 first so that i can update PR 657 continuously? 🙂
Ogali approved PR 672. 🙂

File: src/utils/debug.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ const WARN = 2
 const INFO = 1
 const DEBUG = 0
 /* eslint-enable no-unused-vars */
-let DEBUG_LEVEL = OFF
+let DEBUG_LEVEL = DEBUG
```
[the remark this answers is not in the document]
Comment (ronplaymeta) [reply src/utils/debug.js]: reverted

File: src/utils/objects.js [hunk 2]
```diff
@@ -92,3 +92,16 @@ export function deleteStringValueMatchRecursive(obj, regex) {
...
+
+
+/**
+ * Filter object
+ *
+ * @param {object} obj
+ * @param {Function} callback
+ * @return {object}
+ */
+export function filterObject(obj, callback) {
```
[the remark this answers is not in the document]
Comment (ronplaymeta) [reply src/utils/objects.js:104]: added

File: src/utils/strings.js [hunk 3]
```diff
@@ -73,6 +75,26 @@ export function findUrls(str) {
 }
 
 
+/**
+ * @param {string} str
+ * @param {string} prefix
+ * @return {Array<string>} url matches
+ */
+export function findMarkdownUrls(str, prefix) {
```
[the remark this answers is not in the document]
Comment (ronplaymeta) [reply src/utils/strings.js:83]: added

File: storybook-static/0.34a1ef2c84f7f1b036db.manager.bundle.js
[the remark this answers is not in the document]
Comment (ronplaymeta) [reply storybook-static/0.34a1ef2c84f7f1b036db.manager.bundle.js:1]: reverted

File: src/Components/BranchesControl.jsx
[the remark this answers is not in the document]
Comment (ronplaymeta) [reply src/Components/BranchesControl.jsx:1]: reverted

**Comment by ronplaymeta** [comment]
ptal

File: src/utils/objects.test.js [hunk 4]
```diff
@@ -51,12 +58,48 @@ describe('objects util', () => {
...
         uuids: [undefined, undefined],
       },
     })
   })
+
+  it('filterObject', () => {
+    const testObj = deepCloneObject(templateObj)
+    const filteredObject = filterObject(testObj, (val, key) => {
+      // eslint-disable-next-line no-magic-numbers
+      if (val === 3 && key === 'uuid') {
```
Comment (pablo-mayrgundter) [inline src/utils/objects.test.js:74]: I don't understand this filter condition.

The object filter methods were added to filter out uuids, but the expected value below leaves them in?

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]
One issue with the object unit test.

Also, could you please update the top-level PR description with a link that includes feature flag usage.  I tried `...index.ifc?feature=placemark#f:placemark` but didn't see a placemark

File: src/utils/objects.test.js (hunk 4 above)
Comment (ronplaymeta) [reply src/utils/objects.test.js:74]: Sometimes, parameters behind "?"  are removed before page is loaded, so can't see place mark occasionally.
So, I used hash parameters to not lost place mark feature info.
Test URL is https://deploy-preview-657--bldrs-share.netlify.app/share/v/gh/ronplaymeta/bldrs-ai-Share/main/public/index.ifc#f:placemark

And above filter condition returns a new object that only {'uuid': 3} pairs are removed.

**Review by pablo-mayrgundter — APPROVED** [review]

**Review by pablo-mayrgundter — APPROVED** [review]

**Pull request merged**